river/command.zig

```diff
@@ -117,3 +117,71 @@ pub fn errToMsg(err: Error) [:0]const u8 {
         Error.Other => unreachable,
     };
 }
+
+
+/// args[0] must be the first option if any
+/// offset will be set to the index of the first entry in args that is not an option
+/// Return an error when parsing a value for an option fails
+/// TODO where to enforce the naming of arguments
+fn parse(args: []const []const u8, offset: *usize, comptime T: type) !T {
+    if (@typeInfo(T) != .Struct) @compileError("T must be a struct");
+
+    // The empty {} make sure there is a default value for all members
+    var ret: T = T{};
+    // Make sure all bools are false by default
+    inline for (@typeInfo(T).Struct.fields) |field| {
+        if (@typeInfo(field.field_type) != .Bool) continue;
+        if(field.default_value.?) @compileError("bool options must be false by default");
+
+    }
+    // TODO cleaner way to set offset
+    var i: usize = 0;
+    defer offset.* = i;
+
+    outer: for (args) |arg, idx| {
+        i = idx + 1;
```

Comment:
Pretty sure this is the why the tests are crashing, should just be `i = idx`

Reply:
I changed it to just count the parsed arguments by adding one each time.